internal/http/download_test.go

```diff
@@ -153,7 +205,17 @@ func startFakeDownloadServer(t *testing.T, handler http.Handler) string {
 	serverError := make(chan error)
 	go func() {
 		defer close(serverError)
-		serverError <- server.Serve(listener)
+		if !tls {
+			serverError <- server.Serve(listener)
+			return
+		}
+
+		cert, key := generateSelfSignedCert(t)
+		t.Cleanup(func() {
+			os.Remove(key)
+			os.Remove(cert)
+		})
+		serverError <- server.ServeTLS(listener, cert, key)
```

Comment:
There's also a shorter solution for this in [joinclient_test.go](https://github.com/k0sproject/k0s/blob/v1.31.1%2Bk0s.0/pkg/token/joinclient_test.go#L147-L177). No need to write the certs to disk, so there's no need for cleanups.

Reply:
Way better. This is fixed here.